vivisect/analysis/elf/elfplt_late.py

```diff
@@ -0,0 +1,325 @@
+            # where the magic comes in...
+            funcsz = vw.getFunctionMeta(fva, 'Size')
+            for divisor in range(funcdist, 0, -1):
+                logger.log(e_cmn.SHITE, 'attempting to divide funcdist (0x%x) by %d', funcdist, divisor)
+
+                # does the gap between functions support splitting?
+                if funcdist < (divisor * funcsz):
+                    logger.log(e_cmn.SHITE, '.. not big enough')
+                    continue
+
+                newdist = funcdist // divisor
+                # does the funcdist split evenly?
+                if newdist * divisor != funcdist:
+                    logger.log(e_cmn.SHITE, ".. doesn't divide evenly(newdist: 0x%x  divisor: 0x%x   funcdist: 0x%x)", newdist, divisor, funcdist)
+                    continue
+
+                # do the opcodes support this size split?
+                if not compareFuncs(vw, fva, fva + newdist, funcsz):
+                    logger.log(e_cmn.SHITE, ".. functions don't match!")
+                    continue
+
+                break
+
+            # should end up dividing by 1 if not divisible
+            if divisor > 1:
+                logger.info("dividing our lowest function (0x%x) distance by %d", funcdist, divisor)
+                funcdist //= divisor
+
+            idx = getGoodIndex(curplts, funcdist)
+
+            tmpva = curplts[idx]
+            logger.debug("... backwards from... 0x%x", tmpva)
+            op = vw.parseOpcode(tmpva)
+            stdmnem = op.mnem
+            # start there and go backwards
+            while tmpva > pltva:
+                logger.log(e_cmn.SHITE, "tmpva: 0x%x", tmpva)
+                # check if already in a PLT function (ignore if it's part of some other func)
+                curfunc = vw.getFunction(tmpva)
+                if curfunc is not None and (curfunc == tmpva or isPLT(vw, curfunc)):
+                    #logger.debug('skip 0x%x: already function', tmpva)
+                    tmpva -= funcdist 
+                    continue
+
+                # check if there's enough room for an even additional one beyond this 
+                # (ie. the initial entry in the plt is slightly larger than a normal entry
+                leftovers = tmpva - funcdist - pltva
+                if leftovers != 0 and leftovers < funcdist:
+                    logger.debug('skip 0x%x: too close to beginning of PLT', tmpva)
+                    tmpva -= funcdist 
+                    continue
+                
+                # if standard start of plt mnemonic is different...
+                op = vw.parseOpcode(tmpva)
+                if op.mnem != stdmnem:
+                    logger.debug('skip 0x%x: WRONG MNEM! (is %r   should be: %r)', tmpva, op.mnem, stdmnem)
+                    tmpva -= funcdist 
+                    continue
+
+                logger.info("New PLT Function! 0x%x", tmpva)
+                vw.makeFunction(tmpva)
+
+                tmpva -= funcdist 
```

Comment:
Is it possible that funcdist can be 0?

Reply:
shouldn't be.  look at lines 132-136.  we specifically weed out 0.